### PR TITLE
Door/airlock operation rework (better open/close & door-linked forcefield integration)

### PR DESCRIPTION
**Pull request by Kubius**
[FIX][STATION SYSTEMS][CODE QUALITY]
## About the PR 

- Reworks unpowered_open_close() (the proc for airlock prying) to use the airlock's regular open() and close() procs. These procs now have a manual_actuation variable that, when set, alters them to behave correctly with prying (no checking for power, no airlock operation sound, et cetera).
- Also removes duplicate open and close behavior from pod doors. These now instead have overridden icon/animation procs as necessary (what the previous duplicate behavior was attempting to account for).
- Door-linked forcefield power usage management has been moved to the base door type (since all doors can have linked forcefields), and doors will now be subscribed to process when open if they have a force field that isn't being powered by a door-link shield generator.
- Door-linked forcefields now respond reliably to changes in power; if powered by the door, they will now deactivate if equipment power is lost while the door is open (and reactivate if equipment power is regained), while door-linked shields provided by a shield generator will consistently operate even when area power has failed.

## Why's this needed? 

Makes doors and their forcefields operate in a more cohesive manner (notably including updating adjacent tiles when they're supposed to) and reduces a bit of janky reimplementation.

### CODE REVIEW

File: code/obj/machinery/shield_generators/portable_shield_generator.dm [hunk 1]
```diff
@@ -665,6 +673,7 @@ ADMIN_INTERACT_PROCS(/obj/machinery/shieldgenerator, proc/turn_on, proc/turn_off
 			var/obj/machinery/door/door = (locate() in src.loc)
 			if(door)
 				door.linked_forcefield = src
+				door.SubscribeToProcess() //necessary for power consumption while open
```
Comment (TobleroneSwordfish) [inline code/obj/machinery/shield_generators/portable_shield_generator.dm:676]: This is quite a heavy performance hit if every single door on Nadir is a processing machine, consider making it only subscribe to process when it actually needs to (or just moving power usage to only when the field is created)

File: code/obj/machinery/shield_generators/portable_shield_generator.dm [hunk 2]
```diff
@@ -627,7 +635,7 @@ ADMIN_INTERACT_PROCS(/obj/machinery/shieldgenerator, proc/turn_on, proc/turn_off
 
 
 //sealab arrivalss
-/obj/machinery/door/var/obj/forcefield/energyshield/linked_forcefield = 0
+/obj/machinery/door/var/obj/forcefield/energyshield/linked_forcefield = null
```
Comment (TobleroneSwordfish) [inline code/obj/machinery/shield_generators/portable_shield_generator.dm:638]: This is utterly cursed and shouldn't be here, I charge you with moving it to the `/obj/machinery/door` definition

File: code/obj/machinery/door/airlock.dm [hunk 3]
```diff
@@ -857,44 +833,41 @@ var/global/list/cycling_airlocks = list()
...
-			else
-				D.close()
-
-/obj/machinery/door/airlock/close()
-	//split into two sets of checks so failures to close due to lacking power will cause linked shields to deactivate
-	if ((!src.arePowerSystemsOn()) || (src.status & NOPOWER) || src.isWireCut(AIRLOCK_WIRE_OPEN_DOOR))
-		if (src.linked_forcefield)
-			src.linked_forcefield.setactive(0)
-			power_usage -= LINKED_FORCEFIELD_POWER_USAGE
-		return
```
Comment (TobleroneSwordfish) [inline code/obj/machinery/door/airlock.dm:887]: Removing this early return makes it possible to close doors by hand without power.

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

File: code/obj/machinery/shield_generators/portable_shield_generator.dm (hunk 1 above)
Comment (Kubius) [reply code/obj/machinery/shield_generators/portable_shield_generator.dm:676]: Moving power usage to field creation is a little wonky, but if I can't think of a better way to dodge the processing, I'll probably end up doing that.

File: code/obj/machinery/door/airlock.dm (hunk 3 above)
Comment (Kubius) [reply code/obj/machinery/door/airlock.dm:887]: oops

File: code/obj/machinery/shield_generators/portable_shield_generator.dm (hunk 2 above)
Comment (Kubius) [reply code/obj/machinery/shield_generators/portable_shield_generator.dm]: Done in dfad581

File: code/obj/machinery/door/airlock.dm (hunk 3 above)
Comment (Kubius) [reply code/obj/machinery/door/airlock.dm:887]: fixed in cbe5b1f

File: code/obj/machinery/shield_generators/portable_shield_generator.dm (hunk 1 above)
Comment (Kubius) [reply code/obj/machinery/shield_generators/portable_shield_generator.dm]: Now subscribes to process only when door is open. Also converted poddoors to use base open and close along the way.

**Pull request merged**